subsys/bluetooth/host/hci_ecc.c

```diff
@@ -324,8 +327,22 @@ static void ecc_process(struct k_work *work)
 static void clear_ecc_events(struct net_buf *buf)
 {
 	struct bt_hci_cp_le_set_event_mask *cmd;
+	struct bt_hci_cmd_hdr *hci_cmd_hdr;
+	uint8_t h4_type;
 
-	cmd = (void *)(buf->data + sizeof(struct bt_hci_cmd_hdr));
+	/* This function does not pull from the buffer to keep
+	 * it intact. The buffer is modified in place, and when
+	 * this function returns, the buffer continues to the
+	 * real HCI send function.
+	 */
+
+	__ASSERT_NO_MSG(bt_buf_get_type(buf) == BT_BUF_H4);
+	h4_type = buf->data[0];
+	__ASSERT_NO_MSG(h4_type == BT_HCI_H4_CMD);
+	hci_cmd_hdr = (void *)&buf->data[sizeof(h4_type)];
+	__ASSERT_NO_MSG(hci_cmd_hdr->opcode == BT_HCI_OP_LE_SET_EVENT_MASK);
+
+	cmd = (void *)&buf->data[sizeof(h4_type) + sizeof(*hci_cmd_hdr)];
```

Comment:
We can use `net_buf_clone` to clone the buffer here to a local buffer variable to avoid pulling from the original buffer, while still being able to pull data to avoid stuff like `buf->data[sizeof(h4_type) + sizeof(*hci_cmd_hdr)];`

Reply:
Sounds a bit overkill. What the code here seems to want is to "peek" into the packet and modify a few bits, i.e. not really to parse the packed. Even if you wanted to parse (call e.g. `pull()`) a more light-weight solution than cloning would be to use `net_buf_simple_save/restore()`.